Create template files for documentation (#2)
ansible-community/project-template: docs/references/porting_guide.md

# PROJECTNAME $version Porting guide

> note to developer - consider autogenerating this using 
> [antsibull-changelog](https://pypi.org/project/antsibull-changelog/).  
> If you manually create this, it should include all information a user needs to
> migrate from one version of your project to another, plus all the new features
>  and changes. 
>  The porting guide should link to the release changelogs as well.

Reviewer: ```suggestion
You can automatically generate porting guides with [antsibull-changelog](https://pypi.org/project/antsibull-changelog/).  
If you manually create your porting guides, they should include all information a user needs to migrate from one version of your project to another, plus all the new features and changes. 
Porting guides should also provide a link to release changelogs.
```
